holoviews/core/accessors.py

```diff
@@ -0,0 +1,463 @@
+        params = {p: val for p, val in kwargs.items()
+                  if isinstance(val, param.Parameter)
+                  and isinstance(val.owner, param.Parameterized)}
+        param_methods = {p: val for p, val in kwargs.items()
+                         if util.is_param_method(val, has_deps=True)}
+
+        if dynamic is None:
+            dynamic = (bool(streams) or isinstance(self._obj, DynamicMap) or
+                       util.is_param_method(function, has_deps=True) or
+                       params or param_methods)
+
+        if applies and dynamic:
+            return Dynamic(self._obj, operation=function, streams=streams,
+                           kwargs=kwargs, link_inputs=link_inputs)
+        elif applies:
+            inner_kwargs = dict(kwargs)
+            for k, v in kwargs.items():
+                if util.is_param_method(v, has_deps=True):
+                    inner_kwargs[k] = v()
+                elif k in params:
+                    inner_kwargs[k] = getattr(v.owner, v.name)
+            if hasattr(function, 'dynamic'):
+                inner_kwargs['dynamic'] = False
+            return function(self._obj, **inner_kwargs)
+        elif self._obj._deep_indexable:
+            mapped = []
+            for k, v in self._obj.data.items():
+                new_val = v.apply(function, dynamic=dynamic, streams=streams,
+                                  link_inputs=link_inputs, **kwargs)
+                if new_val is not None:
+                    mapped.append((k, new_val))
+            return self._obj.clone(mapped, link=link_inputs)
+
+        
+    def aggregate(self, dimensions=None, function=None, spreadfn=None, **kwargs):
+        """Applies a aggregate function to all ViewableElements.
+
+        See :py:meth:`Dimensioned.opts` and :py:meth:`Apply.__call__`
+        for more information.
+        """
+        kwargs['_args'] = (dimensions, function, spreadfn)
+        return self.__call__('aggregate', **kwargs)
+
+    def opts(self, *args, **kwargs):
+        """Applies options to all ViewableElement objects.
+
+        See :py:meth:`Dimensioned.opts` and :py:meth:`Apply.__call__`
+        for more information.
+        """
+        kwargs['_args'] = args
+        return self.__call__('opts', **kwargs)
+
+    def reduce(self, dimensions=[], function=None, spreadfn=None, **kwargs):
+        """Applies a reduce function to all ViewableElement objects.
+
+        See :py:meth:`Dimensioned.opts` and :py:meth:`Apply.__call__`
+        for more information.
+        """
+        kwargs['_args'] = (dimensions, function, spreadfn)
+        return self.__call__('reduce', **kwargs)
+
+    def select(self, **kwargs):
+        """Applies a selection to all ViewableElement objects.
+
+        See :py:meth:`Dimensioned.opts` and :py:meth:`Apply.__call__`
+        for more information.
+        """
+        return self.__call__('select', **kwargs)
+
+
+
+class redim(object):
```

Comment:
Any hard in capitalizing this class name to be consistent with the others? (i.e `Redim`)?

Reply:
Maybe pickling issues...could leave an alias around I suppose but maybe it is no longer worth it...

Reply:
Right pickling issues was the reason I left it.

>maybe it is no longer worth it...

To be clear, do you mean renaming it is not worth it?

Reply:
Right. Basically if you think renaming `redim` to `Redim` is easy then it would be good to make things consistent. Otherwise, don't bother.

Reply:
I think I'll file an issue about cleaning this up, for 2.0 we can break imports and I've already made a note to fix some of the inline imports.

Reply:
Sounds good.